homeassistant/components/led_ble/__init__.py

```diff
@@ -0,0 +1,119 @@
+"""The LED BLE integration."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+from led_ble import BLEAK_EXCEPTIONS, LEDBLE
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth.match import ADDRESS, BluetoothCallbackMatcher
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, EVENT_HOMEASSISTANT_STOP, Platform
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEVICE_TIMEOUT, DOMAIN, UPDATE_SECONDS
+from .models import LEDBLEData
+
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Yale Access Bluetooth from a config entry."""
+    address: str = entry.data[CONF_ADDRESS]
+    ble_device = bluetooth.async_ble_device_from_address(hass, address.upper(), True)
+    if not ble_device:
+        raise ConfigEntryNotReady(
+            f"Could not find LED BLE device with address {address}"
+        )
+
+    led_ble = LEDBLE(ble_device)
+
+    @callback
+    def _async_update_ble(
+        service_info: bluetooth.BluetoothServiceInfoBleak,
+        change: bluetooth.BluetoothChange,
+    ) -> None:
+        """Update from a ble callback."""
+        led_ble.set_ble_device(service_info.device)
+
+    entry.async_on_unload(
+        bluetooth.async_register_callback(
+            hass,
+            _async_update_ble,
+            BluetoothCallbackMatcher({ADDRESS: address}),
+            bluetooth.BluetoothScanningMode.PASSIVE,
+        )
+    )
+
+    async def _async_update():
+        """Update the device state."""
+        try:
+            await led_ble.update()
+        except BLEAK_EXCEPTIONS as ex:
+            raise UpdateFailed(str(ex)) from ex
+
+    startup_event = asyncio.Event()
+    cancel_first_update = led_ble.register_callback(lambda *_: startup_event.set())
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=led_ble.name,
+        update_method=_async_update,
+        update_interval=timedelta(seconds=UPDATE_SECONDS),
+    )
+
+    try:
+        await coordinator.async_config_entry_first_refresh()
+    except ConfigEntryNotReady:
+        cancel_first_update()
+        raise
+
+    try:
+        async with async_timeout.timeout(DEVICE_TIMEOUT):
+            await startup_event.wait()
+    except asyncio.TimeoutError as ex:
+        raise ConfigEntryNotReady(
+            "Update sent, but not state received; "
```

Comment:
```suggestion
            "Update sent, but no state received; "
```

Reply:
Maybe even make it more user friendly: 
```suggestion
            "Unable to communicate with the device; "
```